File: services/academic-service/src/main/java/com/visor/school/academic/model/Address.java
package com.visor.school.academic.model;

import jakarta.persistence.Embeddable;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Address value object
 */
@Embeddable
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Address {
    private String street;
    private String city;
    private String state;
    private String postalCode;
    private String country = "Cambodia";

    public String getFullAddress() {
        StringBuilder sb = new StringBuilder();
        sb.append(street);
        if (state != null) {
            sb.append(", ").append(state);
        }
        sb.append(", ").append(city);
        sb.append(" ").append(postalCode);
        if (!"Cambodia".equals(country)) {
            sb.append(", ").append(country);
        }
        return sb.toString();
    }
}

File: services/academic-service/src/main/kotlin/com/visor/school/academicservice/model/Address.kt
package com.visor.school.academicservice.model

import jakarta.persistence.Embeddable

/**
 * Address value object
 */
@Embeddable
data class Address(
    val street: String,
    val city: String,
    val state: String? = null,
    val postalCode: String,
    val country: String = "Cambodia"
) {
    fun getFullAddress(): String {
        return buildString {
            append(street)
            if (state != null) {
                append(", $state")
            }
            append(", $city")
            append(" $postalCode")
            if (country != "Cambodia") {
                append(", $country")
            }
        }
    }
}

